pymeasure/instruments/agilent/agilentE4407B.py

```diff
@@ -0,0 +1,637 @@
+        """ A floating point property that represents the resolution bandwidth
+        in Hz. This property can be set.
+        """,
+    )
+    video_bandwidth = Instrument.control(
+        ":SENS:BAND:VID?",
+        ":SENS:BAND:VID %g",
+        """ A floating point property that represents the video bandwidth
+        in Hz. This property can be set.
+        """,
+    )
+    resolution_bandwidth_auto = Instrument.control(
+        ":SENS:BAND:RES:AUTO?",
+        ":SENS:BAND:RES:AUTO %g",
+        """ A boolean property that represents the resolution bandwidth
+        auto mode. This property can be set.
+        """,
+    )
+    video_bandwidth_auto = Instrument.control(
+        ":SENS:BAND:VID:AUTO?",
+        ":SENS:BAND:VID:AUTO %g",
+        """ A boolean property that represents the video bandwidth
+        auto mode. This property can be set.
+        """,
+    )
+    video_resolution_bandwidth_ratio = Instrument.control(
+        ":SENS:BAND:VID:RAT?",
+        ":SENS:BAND:VID:RAT %g",
+        """ A floating point property that represents the video to resolution
+        bandwidth ratio. This property can be set.
+        """,
+    )
+    video_resolution_bandwidth_ratio_auto = Instrument.control(
+        ":SENS:BAND:VID:RAT:AUTO?",
+        ":SENS:BAND:VID:RAT:AUTO %g",
+        """ A boolean property that represents the video to resolution
+        bandwidth ratio auto mode. This property can be set.
+        """,
+    )
+    detector_auto = Instrument.control(
+        ":SENS:DET:AUTO?",
+        ":SENS:DET:AUTO %g",
+        """ A boolean property that represents the detector auto mode.
+        This property can be set.
+        """,
+    )
+    detector_type = Instrument.control(
+        ":SENS:DET:?",
+        ":SENS:DET %g",
+        """ A string property that represents the detector type.
+        This property can be set.
+        """,
+        validator=strict_discrete_set,
+        values=["NEG", "POS", "SAMPL", "AVER", "RMS"],
+    )
+    average_type = Instrument.control(
+        ":SENS:AVER:TYPE?",
+        ":SENS:AVER:TYPE %g",
+        """ A string property that represents the average type. LPOW is log power
+        and POW is linear power. This property can be set.
+        """,
+        validator=strict_discrete_set,
+        values=["LPOW", "POW"],
+    )
+
+    emi_detector_type = Instrument.control(
+        ":SENS:DET:EMI?",
+        ":SENS:DET:EMI %g",
+        """ A string property that represents the detector type.
+        This property can be set.
+        """,
+        validator=strict_discrete_set,
+        values=["QPE", "AVER", "OFF"],
+    )
+    emi_view_type = Instrument.control(
+        ":SENS:DET:EMI:VIEW?",
+        ":SENS:DET:EMI:VIEW %g",
+        """ A string property that represents the detector type.
+        This property can be set.
+        """,
+        validator=strict_discrete_set,
+        values=["POS", "EMI"],
+    )
+    qp_detector_gain = Instrument.control(
```

Comment:
similar reasoning as above, this should probably be a boolean property `qp_detector_gain_enabled`.